extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +186,120 @@ private void restartJobs() {
+                                createCronForRepeatEverySeconds(thingConfig.getInterval()), start);
+                        Job positionalJob = new PositionalJob(thingUID);
+                        addJobToQueue(positionalJob);
+                        scheduledExecutor.schedule(positionalJob, expression);
+                        logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
+                                thingConfig.getInterval(), thingUID);
+                        
+                        // Execute Positional Job immediately
+                        currentTime = LocalDateTime.now();
+                        cronExpression = new CronExpression(createCronFromTemporal(currentTime));
+                        scheduledExecutor.schedule(positionalJob, cronExpression);
                     }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
                 }
+            } catch (ParseException ex) {
+                logger.error("{}", ex.getMessage(), ex);
+            } finally {
+                monitor.unlock();
             }
-        }, 2000, TimeUnit.MILLISECONDS);
+        }, DELAY, MILLISECONDS);
     }
 
     /**
      * Stops all jobs for this thing.
      */
     private void stopJobs() {
-        logger.debug("Stopping jobs for thing {}", getThing().getUID());
-        synchronized (schedulerLock) {
-            if (quartzScheduler != null) {
-                try {
-                    String thingUid = getThing().getUID().toString();
-                    for (JobKey jobKey : quartzScheduler.getJobKeys(jobGroupEquals(thingUid))) {
-                        logger.debug("Deleting astro {} for thing '{}'", jobKey.getName(), thingUid);
-                        quartzScheduler.deleteJob(jobKey);
-                    }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
-                }
+        ThingUID thingUID = getThing().getUID();
+        logger.debug("Stopping Scheduled Jobs for thing {}", thingUID);
+        monitor.lock();
+        try {
+            if (scheduledExecutor != null) {
+                List<Job> jobsToRemove = scheduledJobs.stream().filter(this::isJobAssociatedWithThing).collect(toList());
+                Consumer<Job> removalFromExecutor = scheduledExecutor::remove;
+                Consumer<Job> removalFromQueue = scheduledJobs::remove;
+                jobsToRemove.forEach(removalFromExecutor.andThen(removalFromQueue));
+                logger.debug("Stopped {} Scheduled Jobs for thing {}", jobsToRemove.size(), thingUID);
             }
+        } catch (final Exception ex) {
+            logger.error("{}", ex.getMessage(), ex);
+        } finally {
+            monitor.unlock();
         }
     }
+    
+    private boolean isJobAssociatedWithThing(Job job) {
+        String thingUID = getThing().getUID().getAsString();
```

Comment:
Why do you get the UID as a string for comparison? `UID` does have an `equals()` method, please use it.